Fix breaking change in paths and runtime warning

oliverprater opened the pull request:
This fixes the faulty commit 23c6f48 and the runtime warnings in consul version >=0.8 due to rpc configuration issues.

In case my suggested fix for the runtime warnings is not well-rounded enough I can open a separate PR with commit 19bfebf

As the incorrect escaping of the string lead to total failure of the consul agent due to:

> Error starting agent: Failed to start Consul server: Error loading CA File: open {{ consul_config_path }}/ssl/ca.pem: no such file or directory

vincent-legoll commented on README.md:
```diff
@@ -213,16 +213,34 @@ in many Ansible versions, so this feature might not always work.
-- Default value: *0.0.0.0*
+- Default value:
+  - consul version  <0.8: *0.0.0.0*
+  - consul verison >=0.8: ""
```
typo "verison" + space around <

vincent-legoll commented on README.md:
```diff
@@ -213,16 +213,34 @@ in many Ansible versions, so this feature might not always work.
-  - `https`: `-1`
-  - `dns`: `8600`
+- The default values are set according to the consul version inside the playbook.
+  - consul verison <0.8:
```
idem + space around <

vincent-legoll commented on README.md:
```diff
@@ -213,16 +213,34 @@ in many Ansible versions, so this feature might not always work.
+    - `http`: `8500`
+    - `https`: `-1`
+    - `rpc`: `8400`
+  - consul verison >=0.8:
```
+1

vincent-legoll commented on tasks/main.yml:
```diff
@@ -15,6 +15,27 @@
+    - consul_ports | length == 0
+    - consul_version | version_compare('0.8', '<')
+
+- name: Set default port configuration for the consul version >=0.8
```
missing space around ">="

vincent-legoll commented on tasks/main.yml:
```diff
@@ -15,6 +15,27 @@
 - name: Include checks/asserts
   include: asserts.yml
 
+- name: Set default port configuration for the consul version <0.8
```
missing space around "<"

vincent-legoll reviewed (commented):
I'm OK with that fix if the small glitches are fixed

vincent-legoll commented on defaults/main.yml:
```diff
@@ -109,7 +114,7 @@ consul_acl_policy: false
 consul_tls_enable: "{{ lookup('env','CONSUL_TLS_ENABLE') | default(false, true) }}"
 consul_tls_src_files: "{{ lookup('env','CONSUL_TLS_SRC_FILES') | default(role_path+'/files', true) }}"
-consul_tls_dir: "{{ lookup('env','CONSUL_TLS_DIR') | default('{{ consul_config_path }}/ssl', true) }}"
+consul_tls_dir: "{{ lookup('env','CONSUL_TLS_DIR') | default(consul_config_path+'/ssl', true) }}"
```
Put spaces around that + operator, please

vincent-legoll commented on templates/config.json.j2:
```diff
@@ -24,8 +24,28 @@
+    {% else %}
+      "ports": {
+        {% if _consul_default_port_rpc | trim != '' %}
+          {% set _ = consul_ports.__setitem__('rpc', _consul_default_port_rpc) %}
```
Why the __setitem__() use ? I'd think this needs at least a comment.

oliverprater replied:
The template uses some jinja2 magic also present in the DebOps project to ensure that the consul_ports dict object is always identical to the config.json file values.

See https://stackoverflow.com/questions/36886650/how-to-add-a-new-entry-into-a-dictionary-object-while-using-jinja2
See https://stackoverflow.com/questions/4870346/can-a-jinja-variables-scope-extend-beyond-in-an-inner-block

Just noticed from my comment it is unclear for what reason we need this jinja2 magic.

When you look, for instance, at `tasks/main.yml` line 238 the variable `{{ consul_ports.http|int }}` is used to `wait_for` the HTTP API to become available. The jinja2 magic ensures that this happens as expected.

oliverprater committed:
FIX: This fixes the broken commit

The issue is that the escaped string will print {{ consul_config_path }} to the file and break the config file.
Error starting agent: Failed to start Consul server: Error loading CA File: open {{ consul_config_path }}/ssl/ca.pem: no such file or directory

oliverprater committed:
Fix the issues with port mappings in consul version greater than 0.8

Moves the default value generation for ports into the template.
The template uses some jinja2 magic also present in the DebOps project to ensure that the consul_ports dict object is always identical to the config.json file values.
See https://stackoverflow.com/questions/36886650/how-to-add-a-new-entry-into-a-dictionary-object-while-using-jinja2
See https://stackoverflow.com/questions/4870346/can-a-jinja-variables-scope-extend-beyond-in-an-inner-block

brianshumate committed:
Merge branch 'master' into master
